sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -363,6 +363,11 @@ abstract class OffsetWindowFunction
    */
   val direction: SortDirection
 
+  /**
+   * Whether the offset is based on the entire frame.
+   */
+  val isWholeBased: Boolean = false
```

Comment:
Just special case the NTH_VALUE aggregate function instead of doing this.

Reply:
I want `NTH_VALUE/FIRST_VALUE/LAST_VALUE` use this.

Reply:
Why would we use this for `FIRST` and `LAST`? How would you deal with `IGNORE NULLS`?

Reply:
`FIRST` and `LAST` is aggregate function, there were PRs who used them as `FIRST_VALUE/LAST_VALUE`.
You can reference https://github.com/apache/spark/pull/25082.
All of LEAD/LAG/NTH_VALUE/FIRST_VALUE/LAST_VALUE need `IGNORE NULLS`, I think we should handle in a consistent way.

Reply:
`LEAD/LAG` not support `IGNORE NULLS` now. Maybe the current implement only considered `postgreSQL`. So I think as long as `NTH_VALUE/FIRST_VALUE/LAST_VALUE` use the same way as `LEAD/LAG`, once the time is right, we can uniformly provide them with support for `IGNORE NULLS`.